src/core/Oscillator.cpp

```diff
@@ -88,6 +118,206 @@ void Oscillator::update( sampleFrame * _ab, const fpp_t _frames,
 	}
 }
 
+void Oscillator::generateSineWaveTable(int bands)
+{
+	bands = bands;
+	m_generatedWaveTable = new sample_t[WAVETABLE_LENGTH];
+	for (int i = 0; i < WAVETABLE_LENGTH; i++)
+	{
+		m_generatedWaveTable[i] = sinf(((float)i / (float)WAVETABLE_LENGTH) * F_2PI);
+	}
+}
+
+void Oscillator::generateSawWaveTable(int bands)
+{
+	float max = 0;
+	m_generatedWaveTable = new sample_t[WAVETABLE_LENGTH];
+	for (int i = 0; i < WAVETABLE_LENGTH; i++)
+	{
+		m_generatedWaveTable[i] = 0.0;
+		for (float n = 1; n <= bands; n++)
+		{
+			m_generatedWaveTable[i] +=powf((float)-1.0, (float)(n + 1)) *
```

Comment:
You can write `(float)-1.0` as `-1.0f`, but instead of `powf((float)-1.0, (float)(n + 1))`, it might be more efficient to write this as `(n%2 ? 1.0f : -1.0f)`, i.e.
```
m_generatedWaveTable[i] += (n%2 ? 1.0f : -1.0f) / n * sinf(F_2PI * (float)(i * n) / (float)WAVETABLE_LENGTH);
```

Reply:
Also, I would provide a comment in each of these functions along the lines of "Compute a band-limited saw wave using its fourier series representation. See https://en.wikipedia.org/wiki/Sawtooth_wave" (or link to some other reference that shows the fourier decomposition of the wave. It seems the Wikipedia page on sawtooth waves conveniently puts that info right at the top).

Otherwise, this just looks like black magic :P

Reply:
Maths updated as requested.

Short descriptions added and links provided to explain the fourier generation.